babyagi.py

```diff
@@ -39,37 +37,46 @@
 YOUR_FIRST_TASK = os.getenv("FIRST_TASK", "")
 assert YOUR_FIRST_TASK, "FIRST_TASK environment variable is missing from .env"
 
-#Print OBJECTIVE
-print("\033[96m\033[1m"+"\n*****OBJECTIVE*****\n"+"\033[0m\033[0m")
+# Print OBJECTIVE
+print("\033[96m\033[1m" + "\n*****OBJECTIVE*****\n" + "\033[0m\033[0m")
 print(OBJECTIVE)
 
-# Configure OpenAI and Pinecone
+# Configure OpenAI
 openai.api_key = OPENAI_API_KEY
-pinecone.init(api_key=PINECONE_API_KEY, environment=PINECONE_ENVIRONMENT)
 
-# Create Pinecone index
+# Create Chroma collection
+chroma_persist_dir = "babyagi"
+chroma_client = chromadb.Client(
+    settings=chromadb.config.Settings(
+        chroma_db_impl="duckdb+parquet",
+        persist_directory=chroma_persist_dir,
+    )
+)
+
 table_name = YOUR_TABLE_NAME
-dimension = 1536
 metric = "cosine"
-pod_type = "p1"
-if table_name not in pinecone.list_indexes():
-    pinecone.create_index(table_name, dimension=dimension, metric=metric, pod_type=pod_type)
-
-# Connect to the index
-index = pinecone.Index(table_name)
+embedding_function = OpenAIEmbeddingFunction(api_key=OPENAI_API_KEY)
```

Comment:
Just thinking it might be better to leave it using the default SentenceTransformerEmbeddingFunction that ships with Chroma. It's another little cost saving for most users

Reply:
Decline